unit-tests/test.cpp

```diff
@@ -13,5 +13,16 @@ INITIALIZE_EASYLOGGINGPP
 #endif
 
 namespace test {
-    std::string context;
+
+
+std::string context;
+bool debug;
+debug_logger log;
+
+
+log_message::~log_message()
+{
+    std::cout << '-' << _type << "- " << _line.str() << std::endl;
```

Comment:
Why do we want to print in the D'tor? we already print at the d() function no?

Reply:
The original though was so someone could do:
```
log_message( info ) << blah << blah;
```
That's still possible -- but not implemented this way, instead doing (to fit with Python syntax, including spaces between args):
```
log.d( blah, blah )
```

Reply:
Sorry still didn't understand.
<< overload concat a string. the final result will be a concated string.
Why the DTor should call "cout".

It means that the user don't need to call log.d() ? and if he will it will be printed twice? once on log.d() and once on the Dtor?